v2/pkg/subscraping/sources/redhuntlabs/redhuntlabs.go

```diff
@@ -0,0 +1,143 @@
+// Package redhuntlabs logic
+package redhuntlabs
+
+import (
+	"context"
+	"errors"
+	"strconv"
+	"strings"
+	"time"
+
+	jsoniter "github.com/json-iterator/go"
+
+	"github.com/projectdiscovery/subfinder/v2/pkg/subscraping"
+)
+
+type Response struct {
+	Subdomains []string         `json:"subdomains"`
+	Metadata   ResponseMetadata `json:"metadata"`
+}
+
+type ResponseMetadata struct {
+	ResultCount int `json:"result_count"`
+	PageSize    int `json:"page_size"`
+	PageNumber  int `json:"page_number"`
+}
+
+type Source struct {
+	apiKeys   []string
+	timeTaken time.Duration
+	errors    int
+	results   int
+	skipped   bool
+}
+
+func (s *Source) Run(ctx context.Context, domain string, session *subscraping.Session) <-chan subscraping.Result {
+	results := make(chan subscraping.Result)
+	s.errors = 0
+	s.results = 0
+	pageSize := 1000
+	go func() {
+		defer func(startTime time.Time) {
+			s.timeTaken = time.Since(startTime)
+			close(results)
+		}(time.Now())
+
+		randomCred := subscraping.PickRandom(s.apiKeys, s.Name())
+		if randomCred == "" || !strings.Contains(randomCred, ":") {
+			s.skipped = true
+			return
+		}
+
+		creds := strings.Split(randomCred, ":")
+		baseUrl := creds[0] + ":" + creds[1]
```

Comment:
storing the API URL as a variable could be a bad idea considering that we have different endpoints based on subscription level (the free one is different, and so are all other plans). A user would have a hard time setting up the URLs in the code and building over the tool once again, so instead we can keep the API endpoint/URL part of the config file so that the user can change it based on the needs and subscription level.

Reply:
Fixed: https://github.com/projectdiscovery/subfinder/pull/978/commits/d32761adb4413988aa3fb8c4cb667e5a12ee211e#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5
Had accidentally added a file named do.txt which was created during testing, it has been removed now.